python/ray/_private/telemetry/open_telemetry_metric_recorder.py

```diff
@@ -23,17 +23,29 @@ class OpenTelemetryMetricRecorder:
     It uses OpenTelemetry's Prometheus exporter to export metrics.
     """
 
```

Comment:
If we go with this approach, suggest we add `Singleton` to the class name to make it obvious. Another solution I also found appealing is using module (method 5 in the stack-overflow post) if we don't care about lazy initialization.